lib/OnboardingMessageProvider.jsm

```diff
@@ -87,22 +105,72 @@ const ONBOARDING_MESSAGES = () => [
   },
   {
     id: "TRAILHEAD_3",
-    template: "trailhead",
     targeting: "trailheadInterrupt == 'cards'",
-    trigger: { id: "firstRun" },
-    includeBundle: {
-      length: 3,
-      template: "onboarding",
-      trigger: { id: "showOnboarding" },
-    },
     utm_term: "trailhead-cards",
+    ...TRAILHEAD_ONBOARDING_TEMPLATE,
   },
   {
     id: "TRAILHEAD_4",
     template: "trailhead",
     targeting: "trailheadInterrupt == 'nofirstrun'",
     trigger: { id: "firstRun" },
   },
+  {
+    id: "TRAILHEAD_5",
+    targeting: "trailheadInterrupt == 'modal_control'",
+    utm_term: "trailhead-modal_control",
+    ...TRAILHEAD_ONBOARDING_TEMPLATE,
+    content: {
+      className: "joinCohort",
+      title: { string_id: "onboarding-welcome-body" },
+      benefits: ["products", "knowledge", "privacy"].map(id => ({
+        id,
+        title: { string_id: `onboarding-benefit-${id}-title` },
+        text: { string_id: `onboarding-benefit-${id}-text` },
+      })),
+      learn: {
+        text: { string_id: "onboarding-welcome-learn-more" },
+        url: "https://www.mozilla.org/firefox/accounts/",
+      },
+      form: {
+        title: { string_id: "onboarding-join-form-header" },
+        text: { string_id: "onboarding-join-form-body" },
+        email: { string_id: "onboarding-join-form-email" },
+        button: { string_id: "onboarding-join-form-continue" },
+      },
+      skipButton: { string_id: "onboarding-start-browsing-button-label" },
+    },
```

Comment:
Might as well extract this out to be like `...TRAILHEAD_MODAL_VARIANT_CONTENT` except maybe `…JOIN_CONTENT` ? Makes it clearer that it's supposed to be the same as TRAILHEAD_1.